src/integration-test/java/uk/gov/ida/integrationtest/TestToolInterfaceMatching/TestToolExamplesSchemaTests.java

```diff
@@ -0,0 +1,611 @@
+import org.opensaml.xmlsec.algorithm.descriptors.SignatureRSASHA1;
+import uk.gov.ida.integrationtest.helpers.MatchingServiceAdapterAppRule;
+import uk.gov.ida.matchingserviceadapter.MatchingServiceAdapterConfiguration;
+import uk.gov.ida.saml.core.extensions.IdaAuthnContext;
+import uk.gov.ida.saml.core.test.builders.AttributeQueryBuilder;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.findAll;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.ida.integrationtest.helpers.AssertionHelper.aMatchingDatasetAssertion;
+import static uk.gov.ida.integrationtest.helpers.AssertionHelper.aSubjectWithAssertions;
+import static uk.gov.ida.integrationtest.helpers.AssertionHelper.anAuthnStatementAssertion;
+import static uk.gov.ida.integrationtest.helpers.RequestHelper.makeAttributeQueryRequest;
+import static uk.gov.ida.saml.core.test.TestEntityIds.HUB_ENTITY_ID;
+import static uk.gov.ida.saml.core.test.builders.AddressAttributeBuilder_1_1.anAddressAttribute;
+import static uk.gov.ida.saml.core.test.builders.AddressAttributeValueBuilder_1_1.anAddressAttributeValue;
+import static uk.gov.ida.saml.core.test.builders.DateAttributeBuilder_1_1.aDate_1_1;
+import static uk.gov.ida.saml.core.test.builders.DateAttributeValueBuilder.aDateValue;
+import static uk.gov.ida.saml.core.test.builders.GenderAttributeBuilder_1_1.aGender_1_1;
+import static uk.gov.ida.saml.core.test.builders.IssuerBuilder.anIssuer;
+import static uk.gov.ida.saml.core.test.builders.PersonNameAttributeBuilder_1_1.aPersonName_1_1;
+import static uk.gov.ida.saml.core.test.builders.PersonNameAttributeValueBuilder.aPersonNameValue;
+
+public class TestToolExamplesSchemaTests {
+    private final SignatureAlgorithm signatureAlgorithmForHub = new SignatureRSASHA1();
+    private final DigestAlgorithm digestAlgorithmForHub = new DigestSHA256();
+    private static final String REQUEST_ID = "default-match-id";
+    private static final String PID = "default-pid";
+    private static final ObjectMapper objectMapper = Jackson.newObjectMapper().setDateFormat(ISO8601DateFormat.getDateInstance());
+    private static final Integer replaceYesterday = 1;
+    private static final Integer replace405to100 = 100;
+    private static final Integer replace405to101 = 150;
+    private static final Integer replace405to200 = 400;
+    private static final Integer replace180to100 = 100;
+    private static final Integer replace180to101 = 140;
+    private static final Integer replace180to150 = 160;
```

Comment:
Replace is a slightly odd name as it looks like we'll be replacing 180 with 150 but the value is 160. Maybe something like `inRange180to150`?